pkg/sql/parser/builtins.go

```diff
@@ -366,6 +366,18 @@ var Builtins = map[string][]Builtin{
 		},
 	},
 
+	"gen_random_uuid": {
+		Builtin{
+			Types:      ArgTypes{},
+			ReturnType: fixedReturnType(TypeBytes),
```

Comment:
Our current `uuid_v4` returns `BYTES` for backwards-compatibility reasons (from before we supported `UUID` columns), it would be really great if newer versions could match Postgres here by returning a `TypeUUID` instead!

In addition, we mark some functions as `impure` so that our planning code knows how to deal with them. Since this function is impure, could you add the `impure` flag? (see [here](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/parser/builtins.go#L2339-L2348)). And while we're at it we can tag it as a `categoryIDGeneration` (which is used for our docs) as well.

Reply:
Thanks for comments! 
You are right about the type - I was confused by the nearby UUID functions.
It should be better by now.